src/hooks/useENS.ts

```diff
@@ -0,0 +1,23 @@
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+import { getProvider } from 'services/web3';
+import { EthNetworks } from 'services/web3/types';
+
+const useENS = (address: string | null | undefined) => {
```

Comment:
Hooks are a way to organize logic outside a react component - but it still runs inside the function component. Which is why I'd always prefer other ways to fetch data. Why isn't this a side effect of redux state? We can have the ensName available as part of app state.

Reply:
Need to try this, I'm good with context api :D will try redux for this 😄

Reply:
I think its but complex for me @zavelevsky maybe we can do a follow up PR 🙇🏻